### PR TITLE
feat(config): add Sunricher SR-ZV2833PAC

### LINKED ISSUE #5850: Missing device configuration: sunricher SR-ZV2833PAC
### Checklist

- [X] It is not in the [configuration DB](https://devices.zwave-js.io/?search=3)

- [X] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)

### Which device is missing?

sunricher SR-ZV2833PAC

### Manufacturer ID

0x0330

### Product Type

0x0400

### Product ID

0xd105

### Firmware Version

1.0.0

### Is the device listed on the Z-Wave Alliance website?

no

### Do you have a manual?

https://www.sunricher.com/media/resources/manual/SR-ZV2833PAC-Instruction.pdf
[debug-info.json.txt](https://github.com/zwave-js/node-zwave-js/files/11581839/debug-info.json.txt)

### PR DESCRIPTION
**Pull request by apella12**
Found this in OpenSmartHouse DB Should close #5850 as help wanted, so giving it a go.  Do not have the device, so can't test

PR description here

### CODE REVIEW

**Comment by apella12** [comment]
I see an issue with the allowManualEntry from the wizard extracted file.  Not sure exactly what that means.  It seems to be picking up the "Limit to Options" flag in the OpenSmartHouse DB.  What that means in OpenHAB terms is only the options shown are allowed, whereas I literally read allowManualEntry as meaning as the opposite, so need to understand better to fix without messing this up for a device owner.

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		"max": "255.255"
+	},
+	"associations": {
+		"1": {
+			"label": "Lifeline",
+			"maxNodes": 5,
+			"isLifeline": true
+		},
+		"2": {
+			"label": "Launch 1",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json:23]: Do we know what those additional associations do? Maybe there's a better name? Or are these just 4 control groups?

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+		"5": {
+			"label": "Launch 4",
+			"maxNodes": 5
+		}
+	},
+	"paramInformation": [
+		{
+			"#": "2",
+			"label": "External Switch Type",
+			"description": "External Switch Type",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json]:
```suggestion
```
We avoid descriptions that duplicate or merely restate the labels.
Same for several others.

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+					"value": 2
+				}
+			]
+		},
+		{
+			"#": "3",
+			"label": "Duration of the Multilevel Start Level Change Up/Down",
+			"description": "Set the duration of the Multilevel Start Level Change Up/Down",
+			"valueSize": 1,
+			"minValue": 0,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json:67]: This needs a unit

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+			"defaultValue": 8
+		},
+		{
+			"#": "4",
+			"label": "Associations in Z-Wave network Security Mode",
+			"description": "Associations in Z-Wave network Security Mode",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 15,
+			"defaultValue": 15
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json]: I guess this is meant to toggle the use of Security for the 4 non-lifeline groups? In this case, this should be split into 4 [partial parameters](https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=partial-parameters), one per association group.

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+				},
+				{
+					"label": "button pressed and held",
+					"value": 4
+				},
+				{
+					"label": "button released",
+					"value": 8
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json]: This looks like they are meant to be combined, so this needs one partial parameter per scene (Pressed 1x, Pressed 2x, Press and hold, Release).
I'm sure there is a similar device already that this definition can more or less be copied from.

Same goes for params 6-8, but they can most likely be `$import`-ed from param 5 -> https://zwave-js.github.io/node-zwave-js/#/config-files/using-templates?id=example-3

File: packages/config/config/devices/0x0330/zv2833pac.json [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+			"label": "Function of the connected switches",
+			"description": "Function of the connected switches",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 2,
+			"defaultValue": 0,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "function 1",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0330/zv2833pac.json]: is there an explanation what functions 1-3 mean?

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by apella12** [comment]
Just a comment before I get into this; This disabuses me of the idea I could rip through 4-5 of these by simply extracting from the Open Smart House DB.  I was starting to suspect that was too easy to be true.  Now I'll have to read the docs ;-)

**Comment by AlCalzone** [comment]
Yeah, we have probably the highest standards of all projects when it comes to config params.

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 1 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json:23]: From the manual posted, these are four control groups that do different things based on the based on the switch type - param 2 and function param 9.  This is a complicated device and the owner is going to have to read the manual several times (IMO)

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 2 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json]: Done

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 3 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json:67]: Done

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 4 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json]: I did my best (at my current level of skill) in the next commit.  Let me know if okay.

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 5 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json]: ditto from comment above

File: packages/config/config/devices/0x0330/zv2833pac.json (hunk 6 above)
Comment (apella12) [reply packages/config/config/devices/0x0330/zv2833pac.json]: Like comment above the functions mean something different based on the association groups and run several paragraphs

**Review by AlCalzone — DISMISSED** [review]
Thanks, I had a few more changes for consistency, but I did them myself.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**